pkg/build/api/types.go

```diff
@@ -936,6 +943,43 @@ type BuildLog struct {
 	unversioned.TypeMeta
 }
 
+// DockerStrategyOptions contains extra strategy options for Docker builds
+type DockerStrategyOptions struct {
+	unversioned.TypeMeta
```

Comment:
A couple reasons, none of which are important.

- `./hack/update-generated-deep-copies.sh` threw an error when it wasn't there. 
- I couldn't find a reason why/why not to have them, and I saw that many similar structs had it.
- Some structs that are created have no fields, which also caused `make update` to fail

I've been able to remove it from DockerStrategyOptions and BuildStrategyOptions, but not from the empty stucts.